pkg/apis/discovery/validation/validation_test.go

```diff
@@ -88,6 +88,30 @@ func TestValidateEndpointSlice(t *testing.T) {
 				}},
 			},
 		},
+		"app-protocols": {
+			expectedErrors: 0,
+			endpointSlice: &discovery.EndpointSlice{
+				ObjectMeta:  standardMeta,
+				AddressType: addressTypePtr(discovery.AddressTypeIP),
+				Ports: []discovery.EndpointPort{{
+					Name:        utilpointer.StringPtr("one"),
+					Protocol:    protocolPtr(api.ProtocolTCP),
+					AppProtocol: utilpointer.StringPtr("HTTP"),
+				}, {
+					Name:        utilpointer.StringPtr("two"),
+					Protocol:    protocolPtr(api.ProtocolTCP),
+					AppProtocol: utilpointer.StringPtr("https"),
+				}, {
+					Name:        utilpointer.StringPtr("three"),
+					Protocol:    protocolPtr(api.ProtocolTCP),
+					AppProtocol: utilpointer.StringPtr("my-protocol"),
+				}},
+				Endpoints: []discovery.Endpoint{{
+					Addresses: generateIPAddresses(1),
+					Hostname:  utilpointer.StringPtr("valid-123"),
```

Comment:
add testcase for `example.com/custom-protocol`

Reply:
Thanks, I added this and addressed your other comments